includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -1932,6 +1995,241 @@ public function add_twentytwenty_toggles() {
+			},
+			11
+		);
+	}
+
+	/**
+	 * Amend the Twenty Twenty-One stylesheet to make it compatible with the changes made to the document during
+	 * sanitization.
+	 */
+	public static function amend_twentytwentyone_styles() {
+		add_action(
+			'wp_enqueue_scripts',
+			static function() {
+				$style_handle = 'twenty-twenty-one-style';
+
+				// Bail if the stylesheet is not enqueued.
+				if ( ! wp_style_is( $style_handle ) ) {
+					return;
+				}
+
+				// Set the registered handle as an alias for other stylesheets to depend on.
+				wp_styles()->registered[ $style_handle ]->src = false;
+
+				$css_file = get_theme_file_path(
+					sprintf( 'style%s.css', is_rtl() ? '-rtl' : '' )
+				);
+
+				if ( ! file_exists( $css_file ) ) {
+					return;
+				}
+
+				$styles = file_get_contents( $css_file ); //phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+
+				// Make the rules for the mobile menu less specific so that it can also be applied to the
+				// amp-lightbox mobile menu also.
+				$new_styles = str_replace( '.primary-navigation > .primary-menu-container', '.primary-navigation .primary-menu-container', $styles );
+				$new_styles = str_replace( '.primary-navigation > div > .menu-wrapper', '.primary-navigation div > .menu-wrapper', $new_styles );
+
+				// Append any extra rules that may be needed.
+				$new_styles .= '
+					.primary-navigation-open .menu-button-container {
+						/* Needs to be shown above amp-lightbox, which has a z-index of 1000. */
+						z-index: 1001;
+					}
+
+					/* Hide the desktop menu so that it does not interfere when the amo-lightbox mobile menu is open. */
+					@media only screen and (max-width: 481px) {
+						.primary-navigation > .primary-menu-container {
+							display: none;
+						}
+					}
+
+					/* Show the sub-menu on hover of menu item */
+					.primary-navigation div > .menu-wrapper .menu-item-has-children:hover > .sub-menu {
+						display: block;
+					}
+
+					/* Hide the plus icon on hover of menu item */
+					.primary-navigation div > .menu-wrapper .menu-item-has-children:hover > .sub-menu-toggle > .icon-plus {
+						display: none;
+					}
+
+					/* Show the minus icon on hover of menu item */
+					.primary-navigation div > .menu-wrapper .menu-item-has-children:hover > .sub-menu-toggle > .icon-minus {
+						display: flex;
+					}
+				';
+
+				wp_add_inline_style( $style_handle, $new_styles );
+			},
+			11
+		);
+	}
+
+	/**
+	 * Make the dark mode toggle in the Twenty Twenty-One theme AMP compatible.
+	 *
+	 * Note: This is only shown within the Customizer preview for now, as there is no feasible way of persisting and
+	 * unserializing the user's preference when they switch to dark (or light) mode.
+	 */
+	public function add_twentytwentyone_dark_mode_toggle() {
+		$button = $this->dom->getElementById( 'dark-mode-toggler' );
+
+		if ( ! $button ) {
+			return;
+		}
+
+		$button->setAttribute( Attribute::STYLE, 'display: block;' );
```

Comment:
Would this be better rather as a new `style` element as opposed to an inline style? That will reduce the size since it won't need specificity hacks.

```suggestion
		$style = $this->dom->createElement( 'style' );
		$style->textContent = "#dark-mode-toggler { display: block; }";
		$this-dom->head->appendChild( $style );
```